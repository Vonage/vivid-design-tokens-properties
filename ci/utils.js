import fs from 'fs';

export function clearFolder(path) {
	fs.rmdirSync(path, { recursive: true });
}

export function copySorted(source, target) {
	if (!source || typeof source !== 'object') {
		throw new Error(`source parameter MUST be an non-null object; got '${source}'`);
	}
	if (!target || typeof target !== 'object') {
		throw new Error(`target parameter MUST be an non-null object; got '${target}'`);
	}

	Object.keys(source).sort().forEach(key => {
		let s = source[key];
		if (s && typeof s === 'object') {
			s = copySorted(s, {});
		}
		target[key] = s;
	});

	return target;
}

export function writeJson(data, path) {
	const folder = path.replace(/\/[^\/]+\.[\w]+$/, '');
	fs.mkdirSync(folder, { recursive: true });

	const output = JSON.stringify(data, null, 2);
	fs.writeFileSync(path, output, { encoding: 'utf-8' });
}