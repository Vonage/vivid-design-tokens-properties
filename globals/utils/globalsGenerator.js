function generateAlias(category, name, level, property) {
    return `{alias.${category}.${name}.${level}.${property}.value}`;
}

export default function generateTokens({
                                           category,
                                           type,
                                           flavours,
                                           properties,
                                           aliasFunction = generateAlias,
                                       }) {
    return flavours.reduce((tokensSkeletonOutput, level) => {
        tokensSkeletonOutput[type][level] = properties.reduce((levelProps, property) => {
            levelProps[property] = {
                value: aliasFunction(category, type, level, property)
            }

            return levelProps;
        }, {});
        return tokensSkeletonOutput;
    }, {
        [type]: {}
    });
}
