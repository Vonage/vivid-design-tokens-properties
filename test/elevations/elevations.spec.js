import elevationShadowTokens from '../../globals/shadow/elevation';
import elevationColorTokens from '../../globals/color/elevation';
import StyleDictionaryPackage from 'style-dictionary';
import {getStyleDictionaryConfig} from "../utils";

import elevationParser from '../../ci/parsers/elevation-data-parser.js';
import { rawData } from "./elevation-test-data.js";

describe(`Elevation Tokens`, function () {
    it(`should create the shadow tokens`, function () {
        expect(JSON.stringify(elevationShadowTokens, null, 2)).toMatchSnapshot();
    });

    it(`should create the color tokens`, function () {
        expect(JSON.stringify(elevationColorTokens, null, 2)).toMatchSnapshot();
    });

    // it(`should transpile in styledDictionary`, function () {
    //     const log = console.log;
    //     console.log = jest.fn();
    //     for (const scheme of ['dark', 'light']) {
    //         for (const alt of ['alternate', 'main']) {
    //             const config = getStyleDictionaryConfig([
    //                 `./globals/elevations/tokens.js`,
    //                 `./elevations/${scheme}/${alt}.json`
    //             ], 'elevations.scss');
    //             expect(() => StyleDictionaryPackage.extend(config).buildPlatform('web')).not.toThrow();
    //         }
    //     }
    //     console.log = log;
    // });
    //
    // it(`should generate the correct elevations values json from Figma`, function () {
    //     function mockWriteJson(data, path) {
    //         results.push({
    //             data, path
    //         });
    //     }
    //
    //     const results = [];
    //     elevationParser.parse(rawData, mockWriteJson);
    //
    //     expect(JSON.stringify(results, null, 2)).toMatchSnapshot();
    // });
});
