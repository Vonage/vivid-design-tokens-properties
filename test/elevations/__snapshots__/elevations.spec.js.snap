// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elevation Tokens should create the color tokens 1`] = `
"{
  \\"color\\": {
    \\"elevation\\": {
      \\"dp-2\\": {
        \\"canvas\\": {
          \\"value\\": \\"{alias.color.elevation.dp-2.canvas.value}\\"
        }
      },
      \\"dp-4\\": {
        \\"canvas\\": {
          \\"value\\": \\"{alias.color.elevation.dp-4.canvas.value}\\"
        }
      },
      \\"dp-8\\": {
        \\"canvas\\": {
          \\"value\\": \\"{alias.color.elevation.dp-8.canvas.value}\\"
        }
      },
      \\"dp-12\\": {
        \\"canvas\\": {
          \\"value\\": \\"{alias.color.elevation.dp-12.canvas.value}\\"
        }
      },
      \\"dp-16\\": {
        \\"canvas\\": {
          \\"value\\": \\"{alias.color.elevation.dp-16.canvas.value}\\"
        }
      },
      \\"dp-24\\": {
        \\"canvas\\": {
          \\"value\\": \\"{alias.color.elevation.dp-24.canvas.value}\\"
        }
      }
    }
  }
}"
`;

exports[`Elevation Tokens should create the shadow tokens 1`] = `
"{
  \\"shadow\\": {
    \\"elevation\\": {
      \\"dp-2\\": {
        \\"filter\\": {
          \\"value\\": \\"{alias.shadow.elevation.dp-2.filter.value}\\"
        }
      },
      \\"dp-4\\": {
        \\"filter\\": {
          \\"value\\": \\"{alias.shadow.elevation.dp-4.filter.value}\\"
        }
      },
      \\"dp-8\\": {
        \\"filter\\": {
          \\"value\\": \\"{alias.shadow.elevation.dp-8.filter.value}\\"
        }
      },
      \\"dp-12\\": {
        \\"filter\\": {
          \\"value\\": \\"{alias.shadow.elevation.dp-12.filter.value}\\"
        }
      },
      \\"dp-16\\": {
        \\"filter\\": {
          \\"value\\": \\"{alias.shadow.elevation.dp-16.filter.value}\\"
        }
      },
      \\"dp-24\\": {
        \\"filter\\": {
          \\"value\\": \\"{alias.shadow.elevation.dp-24.filter.value}\\"
        }
      }
    }
  }
}"
`;
